helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -199,20 +238,13 @@ public ZkBaseDataAccessor(String zkAddress, ZkSerializer zkSerializer,
    * @param pathBasedZkSerializer
    * @param zkClientType
    */
-  public ZkBaseDataAccessor(String zkAddress, PathBasedZkSerializer pathBasedZkSerializer,
+  public ZkBaseDataAccessor(String zkAddress,
```

Comment:
I think we should make the behavior description in the Java doc.

Reply:
As discussed offline, we don't want the fallback behavior and also make these old constructors deprecated.